mdm/checkin.go

```diff
@@ -23,6 +23,12 @@ type Authenticate struct {
 	Topic string
 	Raw   []byte `plist:"-"` // Original Authenticate XML plist
 
+	// Additional fields required in AuthenticateRequest as specified
+	// in the Apple documentation.
+	DeviceName string
```

Comment:
It looks like these fields could be stored in the db in the `nano_devices` table. See https://github.com/fleetdm/nanomdm/blob/apple-mdm/storage/mysql/mysql.go#L101 for were the authenticate request is used.

Should we store this additional info?

Reply:
It's a good thought although not something that I needed at the moment as I just wanted to be able to get these fields without recreating/copying the nanomdm XML decoders. I figured I would keep the changes as lightweight as possible and avoid migrations.

Reply:
I don't think any additional work would be required to decode the xml. If you are able to read these fields, then that means the xml decoding was successful.

Reply:
we're going to store these in `hosts`, is there a reason for having them in `nano_devices` too? if not I vote to merge as-is and add to the table if/when we need it!

Reply:
> I don't think any additional work would be required to decode the xml. If you are able to read these fields, then that means the xml decoding was successful.

I think we are on the same page. Adding these to the struct was all that I needed for the XML [decoder](https://github.com/micromdm/nanomdm/blob/main/mdm/checkin.go#L149) to give me the additional request info that I wanted.

Reply:
In order to use these fields from fleet without storing them, you would have to modify the nano check in handler so that you could add custom logic to handle the authenticate request. If we store them in the `nano_devices` table, that would let fleet read the information from that table later.

Reply:
right, I think you're missing the context of the sibling PR to the fleet repo, we're grabbing these fields in a middleware where we do the serial number matching, and we enhance the row with the hardware uuid.

Reply:
fwiw, I think that saving the fields in the `nano_devices` table and then retrieving from that table shortly after is also a valid approach, probably worth having this discussion in that PR with all the context.